DependencyInjection/Compiler/CacheCompatibilityPass.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Bundle\MongoDBBundle\DependencyInjection\Compiler;
+
+use Doctrine\Bundle\MongoDBBundle\DependencyInjection\DoctrineMongoDBExtension;
+use Doctrine\Common\Cache\CacheProvider;
+use Doctrine\Common\Cache\Psr6\CacheAdapter;
+use Doctrine\Common\Cache\Psr6\DoctrineProvider;
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\DependencyInjection\ChildDefinition;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+
+use function array_keys;
+use function assert;
+use function is_a;
+use function is_string;
+use function trigger_deprecation;
+
+/** @internal */
+final class CacheCompatibilityPass implements CompilerPassInterface
+{
+    private const CACHE_SETTER_METHODS_PSR6_SUPPORT = [
+        'setMetadataCache' => true,
```

Comment:
Not sure if I should include both of these methods, and also not sure if I have missed any methods. These were the only ones I could think of.

Reply:
In this case, we only need to do this for `setMetadataCache`: the extension always configures a call to `setMetadataCache`, regardless of the type of cache involved. The compiler pass then checks the method argument and decides whether the argument needs to be wrapped in an adapter.

If we bump the required version of ODM to 2.3 (which we should do in this PR), we don't have to worry about the method not being there at all, so we can stop using the deprecated API immediately.

Reply:
++ 
Also bumped version of ODM to 2.3